### PR TITLE
move Tar Dataset to Tar DataPipe

### PR DESCRIPTION
**Pull request by glaringlee**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#51398 move Tar Dataset to Tar DataPipe**

This PR is to make previously called TarDataset to Tar DataPipe and apply the new model structure.

Differential Revision: [D26162319](https://our.internmc.facebook.com/intern/diff/D26162319)

### CODE REVIEW

**Review by ejguan — APPROVED** [review]
LGTM

File: torch/utils/data/datapipes/iter/readfilesfromtar.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
+from torch.utils.data.dataset import IterableDataset as IterDataPipe
```
Comment (ejguan) [inline torch/utils/data/datapipes/iter/readfilesfromtar.py:1]: I find an easy way just by aliasing `IterableDataset` as `IterDataPipe` in the `__init__.py` under data, then directly import IterDataPipe.

Comment (ejguan) [reply torch/utils/data/datapipes/iter/readfilesfromtar.py:1]: PR #51488

Comment (glaringlee) [reply torch/utils/data/datapipes/iter/readfilesfromtar.py]: Great! Updated

**Pull request closed without merging**